### PR TITLE
Feat/swarinfo 352 use hashmap for faster lookup

### PR DESCRIPTION
**Pull request by xgroleau**

### CODE REVIEW

File: src/bittybuzz/src/BittyBuzzClosureRegister.cpp [hunk 1]
```diff
@@ -3,18 +3,21 @@
...
+    m_closureHeapIdx(closureHeapIdx),
+    m_selfHeapIdx(selfHeapIdx),
+    m_description(std::move(description)) {}
 
 bool BittyBuzzClosureRegister::registerClosure(const char* functionName,
                                                bbzheap_idx_t closureHeapIdx,
                                                bbzheap_idx_t selfHeapIdx,
                                                const BittyBuzzFunctionDescription& description) {
-    if (m_closureRegistersLength >= m_maxSize) {
+    if (m_closureNameRegisters.isFull() || m_closureRegisterMap.isFull()) {
```
Comment (t-constantin) [inline src/bittybuzz/src/BittyBuzzClosureRegister.cpp:20]: To we want to keep older values or insert newer ones?

File: src/bittybuzz/src/BittyBuzzClosureRegister.cpp [hunk 2]
```diff
@@ -29,49 +32,38 @@ bool BittyBuzzClosureRegister::registerClosure(const char* functionName,
...
-
-    for (uint16_t i = 0; i < m_closureRegistersLength; i++) {
-        size_t hash = std::get<0>(m_closureRegisters[i]);
-        if (functionNameHash == hash) {
-            return std::get<1>(m_closureRegisters[i]);
-        }
-    }
-
-    return {};
+    return m_closureRegisterMap.at(functionName);
```
Comment (t-constantin) [inline src/bittybuzz/src/BittyBuzzClosureRegister.cpp:55]: usually a hashmap throws when using at with somehting not in the map, is this the case for a cpp-common/HashMapStack ?

File: src/bittybuzz/src/BittyBuzzClosureRegister.cpp [hunk 3]
```diff
@@ -29,49 +32,38 @@ bool BittyBuzzClosureRegister::registerClosure(const char* functionName,
...
 }
 
 std::optional<std::reference_wrapper<const BittyBuzzRegisteredClosure>> BittyBuzzClosureRegister::
     getRegisteredClosure(uint16_t idx) const {
-    if (idx > m_closureRegistersLength - 1) {
-        return {};
+    auto functionNameOpt = m_closureNameRegisters.at(idx);
```
Comment (t-constantin) [inline src/bittybuzz/src/BittyBuzzClosureRegister.cpp:60]: idem as last comment

File: src/bittybuzz/src/BittyBuzzClosureRegister.cpp [hunk 4]
```diff
@@ -29,49 +32,38 @@ bool BittyBuzzClosureRegister::registerClosure(const char* functionName,
...
+    if (functionNameOpt) {
+        return m_closureRegisterMap.at(functionNameOpt.value().get());
```
Comment (t-constantin) [inline src/bittybuzz/src/BittyBuzzClosureRegister.cpp:62]: idem

**Review by t-constantin — APPROVED** [review]
Few questions but more of my understanding than the actual code.

File: src/bittybuzz/src/BittyBuzzClosureRegister.cpp (hunk 2 above)
Comment (xgroleau) [reply src/bittybuzz/src/BittyBuzzClosureRegister.cpp:55]: Nope, it returns an optional on our implementation

File: src/bittybuzz/src/BittyBuzzClosureRegister.cpp (hunk 1 above)
Comment (xgroleau) [reply src/bittybuzz/src/BittyBuzzClosureRegister.cpp:20]: Removed so we can overwrite existing even if the hashmap is full

**Pull request merged**